psydac/fem/tensor.py

```diff
@@ -894,7 +894,7 @@ def reduce_grid(self, axes=(), knots=()):
                 global_starts[axis][0] = 0
 
         cart = v._cart.reduce_grid(global_starts, global_ends)
-        V    = TensorFemSpace(*spaces, cart=cart, quad_order=self._quad_order)
+        V    = TensorFemSpace(*spaces, cart=cart, quad_order=self._quad_order, dtype=v.dtype)
```

Comment:
The method `reduce_grid` of the `TensorFemSpace`  don't seem to be used and the method `reduce_grid` of the `CartDecomposition` is only used in the `TensorFemSpace` method.

For me, Those two methods are not used anywhere in the library and can be removed.

Reply:
`reduce_grid `is used for the multigrid method, I'm not sure if it works anymore, but you can delete it if you want to

Reply:
Well this could still be useful in the future!

@tomcaruso Could you check if this function works after changing the arguments? Please ask @saidctb if you need any clarification.

Reply:
I failed to make the `CartDecomposition.reduce_grid` worked but otherwise, I found a way to adapt the `TensorFemSpace.reduce_grid()`

Reply:
I crated an issue #311 for this problem. Maybe @saidctb  you should check it.